libpod/container_attach.go

```diff
@@ -81,11 +85,19 @@ func (c *Container) attachContainerSocket(resize <-chan remotecommand.TerminalSi
 			logrus.Warnf("Failed to write to control file to resize terminal: %v", err)
 		}
 	})
-	logrus.Debug("connecting to socket ", c.AttachSocketPath())
 
-	conn, err := net.DialUnix("unixpacket", nil, &net.UnixAddr{Name: c.AttachSocketPath(), Net: "unixpacket"})
+	socketPath := c.AttachSocketPath()
+
+	maxUnixLength := int(C.unix_path_length())
+	if maxUnixLength < len(socketPath)-1 {
+		socketPath = socketPath[0:maxUnixLength]
```

Comment:
Do we need to subtract one from maxUnixLength here to account for a trailing null?

Reply:
thanks for catching it.  No it is not needed, I've tried both versions and I am surprised it doesn't crash.  I've dropped the `-1` and pushed a new version